src/main/java/xbot/common/trajectory/SimpleTimeInterpolator.java

```diff
@@ -137,17 +156,30 @@ public InterpolationResult calculateTarget(Translation2d currentLocation) {
             targetKeyPoint = keyPoints.get(index);
             log.info("Baseline is now " + baseline.getTranslation2d()
                     + " and target is now " + targetKeyPoint.getTranslation2d());
+
+            if (mode == SwerveSimpleTrajectoryMode.KinematicsForIndividualPoints || mode == SwerveSimpleTrajectoryMode.KinematicsForPointsList) {
```

Comment:
⭐ you do this check so many times, consider refactoring out a little helper method, maybe something like `boolean requiresCalculator(mode)`?